### ISSUE TITLE
Unable to import medical images

### ISSUE THREAD

**Issue by piiq** [tkeskita/BVtkNodes#10]
Hi again!

With the update I have managed to install the add-on with the version of VTK that comes from pip and the node editor works just fine. Thanks for solving the https://github.com/tkeskita/BVtkNodes/issues/1#issue-495437825 issue

But now I am having difficulties importing medical image data and setting up workflows in blender.
I have tried opening DICOM and NIFTI files and recreating the pipeline I previously coded in vtk-python.

So in python to end up with a mesh from a dicom stack I would normally do something like this:

```python
PathDicom = "files/dicom/sample/" 
reader = vtk.vtkDICOMImageReader()
reader.SetDirectoryName(PathDicom)
reader.Update()

threshold = vtk.vtkImageThreshold ()
threshold.SetInputConnection(reader.GetOutputPort())
threshold.ThresholdByLower(float(300))  # threshold by 300 Hounsfield units
threshold.ReplaceInOn()
threshold.SetInValue(0)  # set all values below 300 to 0
threshold.ReplaceOutOn()
threshold.SetOutValue(1)  # set all values above 300 to 1
threshold.Update()

dmc = vtk.vtkDiscreteMarchingCubes()
dmc.SetInputConnection(threshold.GetOutputPort())
dmc.GenerateValues(1, 1, 1)
dmc.Update()
```

The output port of the marching cubes gives me a mesh.

I have tried to recreate this workflow with BVtkNodes and ran into several bumps right from the beginning.

1. Each time I try to run the node graph the first error says there is no filename or dicom file provided
![image](https://user-images.githubusercontent.com/11668535/79064240-e5845f00-7caf-11ea-85c7-7a67ff7fda51.png)
The NIFTI image reader generates the same error.
Here I attach 2 samples (a dicom and a nifti image of a vertebra) that I am using for this test.

[SampleDICOM.zip](https://github.com/tkeskita/BVtkNodes/files/4466085/SampleDICOM.zip)
[SampleNII.nii.gz](https://github.com/tkeskita/BVtkNodes/files/4466086/SampleNII.nii.gz)

2. In the nodes themselves I am missing some functions that are a part of the filter. For example the `vtkImageTheshold` is missing `ThresholdByLower()` setting and `ReplaceInOn` and `ReplaceOutOn` settings. I assume I can access them via node custom code, is that correct? If yes, can you provide a little bit of guidance on how to do that?

Thank you for developing this add-on.

**Comment by tkeskita** [comment]
Hello @piiq 

There was a bug about handling of attributes whose name contains string "FileName" which I have now fixed. Please download and install latest version and try again. As long as you provide DirectoryName and no FileName, you should be able to get data. You can add an "Info" node after reader to see what data it produces.

As for the second issue, yes, you can add custom code  by selecting node --> go to Properties tab --> Edit Custom Code. Edit custom code in Text Editor, then in Properties tab --> Save Custom Code will save your code to the active node, and it will show up in the node.

Also, please have a look at [https://bvtknodes.readthedocs.io/en/latest/](documentation)

Hope this helps!

**Comment by piiq** [comment]
Hi @tkeskita thanks. This works now with DICOMs. 
I was testing the same workflow with NIFIT images (it's jist a slightly different file container than DICOM) and since the container holds the image in a single file there is no DirectoryName input.
I have the FileName value to be the absolute path the absolute path to the NIFTI and then when I attached the info node I've got the following error:

![image](https://user-images.githubusercontent.com/11668535/79638863-8bdcd280-8190-11ea-8538-132f2f8802d8.png)

Is there anything I am doing wrong here? It seems to me that I am doing the same as the introduction tutorial is suggesting to do with the .vdi image.
